src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -186,6 +210,8 @@ export const executeQueries = createAsyncThunk<
       )
     );
   }
+
+  // Wait for all queries to complete
   await Promise.all(promises);
```

Comment:
should this be all settled?

Reply:
It does not need. Let me take care of this with the below abort issue together.